git.go

```diff
@@ -101,11 +101,11 @@ func doMirrorOrUpdate(gitModule GitModule, workDir string, retryCount int) bool
 	isInModulesCacheDir := strings.HasPrefix(workDir, config.ModulesCacheDir)
 
 	needSSHKey := true
-	if len(gitModule.privateKey) == 0 || strings.Contains(gitModule.git, "github.com") || gitModule.useSSHAgent == true {
+	if len(gitModule.privateKey) == 0 || strings.Contains(gitModule.git, "github.com") || gitModule.useSSHAgent == false {
 		if isControlRepo {
-			needSSHKey = true
-		} else {
 			needSSHKey = false
+		} else {
```

Comment:
I admit the the variable name `needSSHKey` can be misleading. I renamed it to `explicitlyLoadSSHKey` in the next version. Hopefully this helps understanding the logic.